### PR TITLE
Add new initialization

### PR DESCRIPTION
**Pull request by alvarmaciel**
With this PR we can use the new initialization 
https://mermaid.js.org/intro/#installation

### CODE REVIEW

File: sphinxcontrib/mermaid.py [hunk 1]
```diff
@@ -415,15 +415,24 @@ def install_js(
...
     else:
-        _mermaid_js_url = f"https://unpkg.com/mermaid@{app.config.mermaid_version}/dist/mermaid.min.js"
+        _mermaid_js_url = f"https://cdn.jsdelivr.net/npm/mermaid@{app.config.mermaid_version}/dist/mermaid.esm.min.mjs"
     if _mermaid_js_url:
-        app.add_js_file(_mermaid_js_url, priority=app.config.mermaid_js_priority)
+        # Configuration for Mermaid initialization
+        mermaid_init_script = (
+            f"import mermaid from '{_mermaid_js_url}';\n"
+            "let config = { startOnLoad: true };\n"
+            "mermaid.initialize(config);"
```
Comment (mgaitan) [inline sphinxcontrib/mermaid.py:431]: we should support customization via `app.config.mermaid_init_js` here. change the default if needed (update the README)

File: tests/test_html.py [hunk 2]
```diff
@@ -78,17 +84,20 @@ def test_mermaid_init_js(index):
 @pytest.mark.sphinx("html", testroot="markdown")
 def test_html_raw_from_markdown(index):
     assert (
-        '<script src="https://unpkg.com/mermaid@10.2.0/dist/mermaid.min.js"></script>'
+        '<script type="module">import mermaid from '
+        "'https://cdn.jsdelivr.net/npm/mermaid@10.2.0/dist/mermaid.esm.min.mjs';\n"
```
Comment (mgaitan) [inline tests/test_html.py:88]: please upgrade the default version to the 10.6.1, the current latest

File: sphinxcontrib/mermaid.py [hunk 3]
```diff
@@ -415,15 +415,24 @@ def install_js(
     if doctree and not doctree.next_node(mermaid):
         return
 
+    # Add required JavaScript
     # Add required JavaScript
     if not app.config.mermaid_version:
         _mermaid_js_url = None  # assume it is local
     elif app.config.mermaid_version == "latest":
-        _mermaid_js_url = "https://unpkg.com/mermaid/dist/mermaid.min.js"
+        _mermaid_js_url = "https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.esm.min.mjs"
```
Comment (mgaitan) [inline sphinxcontrib/mermaid.py:423]: update the readme in  mermaid_version where the CDN is mentioned

File: sphinxcontrib/mermaid.py [hunk 4]
```diff
@@ -234,24 +234,24 @@ def _render_mm_html_raw(
     self, node, code, options, prefix="mermaid", imgcls=None, alt=None
 ):
     if "align" in node and "zoom_id" in node:
-        tag_template = """<div align="{align}" id="{zoom_id}" class="mermaid align-{align}">
+        tag_template = """<pre align="{align}" id="{zoom_id}" class="mermaid align-{align}">
```
Comment (mgaitan) [inline sphinxcontrib/mermaid.py:237]: the change of the element type, doesn't affect when zoom is enabled? check mermaid_d3_zoom

Comment (alvarmaciel) [reply sphinxcontrib/mermaid.py:237]: Yes, it affects. Next commit with the fix

**Comment by maurerle** [comment]
I currently get the problem of: `Uncaught ReferenceError: mermaid is not defined`

This broke some day in the last few months.

https://assume.readthedocs.io/en/latest/market_config.html

I tested this fix, which worked smoothly.
It would be very good to have this on the main branch.

**Comment by maurerle** [comment]
@mgaitan do you have time to look into this and release this eventually?

**Comment by mgaitan** [comment]
@maurerle sure! 

the only blocker AFAIK is the zoom feature, that's broken in this branch. 

@alvarmaciel is a good friend, his on holidays but I will coordinate with him ASAP.

**Comment by maurerle** [comment]
@alvarmaciel sorry to bump this issue - but could you take a look at the broken zoom feature?
Would be cool to have this fixed directly in pypi :)

**Pull request merged**